nucypher/blockchain/eth/agents.py

```diff
@@ -288,6 +288,22 @@ def get_all_stakes(self, staker_address: str):
         for stake_index in range(stakes_length):
             yield self.get_substake_info(staker_address=staker_address, stake_index=stake_index)
 
+    @validate_checksum_address
+    def is_restaking(self, staker_address: str) -> bool:
+        staker_info = self.get_staker_info(staker_address)
+        restake_flag = bool(staker_info[3])
```

Comment:
What do you think about named constants for these index numbers?...Same comment for line 304.

Reply:
Hmm - let's open an Issue for this proposal.

Reply:
Yes, not only name constants (or other similar approach), but also make sure access to individual slots of contracts' structs (e.g., `StakerInfo`) is correct. This is precisely what the `verifyState()` discovered and the reason why we couldn't upgrade `StakingEscrow` yesterday (due to a change in the struct).

Reply:
I hope this someday this structure will be private field and will be separate getters. But in any case this order of fields will be the same all the time so it's good to have as a constant in python.
Also want to say - number of field is not fully related to slots. Because slot is just contract state thing (multiple value could be in one slot) and number of field come from ABI of specific function

Reply:
Yes, I think when we were talking about slots we weren't referring to the contract state slots, but the variables in the struct.

Reply:
Issue #1348 :heavy_check_mark: